int32 intensity_input
